corehq/apps/userreports/views.py

```diff
@@ -1110,6 +1110,36 @@ class CreateDataSourceView(BaseEditDataSourceView):
     page_title = ugettext_lazy("Create Data Source")
 
 
+class CloneDataSourceView(CreateDataSourceView):
+    urlname = 'clone_configurable_data_source'
+
+    @property
+    def config_id_to_clone(self):
+        return self.kwargs.get('config_id_to_clone')
+
+    def page_url(self):
+        return reverse(self.urlname, kwargs=dict(domain=self.domain, config_id_to_clone=self.config_id_to_clone))
+
+    @property
+    def config(self):
+        data_source_to_clone = get_datasource_config_or_404(self.config_id_to_clone, self.domain)[0]
+        return DataSourceConfiguration(
+            domain=self.domain,
+            referenced_doc_type=data_source_to_clone.referenced_doc_type,
+            description=data_source_to_clone.description,
+            base_item_expression=data_source_to_clone.base_item_expression,
+            configured_filter=data_source_to_clone.configured_filter,
+            configured_indicators=data_source_to_clone.configured_indicators,
+            named_expressions=data_source_to_clone.named_expressions,
+            named_filters=data_source_to_clone.named_filters,
+            display_name='{} - cloned {}'.format(
+                data_source_to_clone.display_name,
+                datetime.datetime.utcnow().strftime('%Y-%m-%d')
+            ),
+            backend_id=data_source_to_clone.backend_id
+        )
```

Comment:
the better approach to cloning json objects is to do this `new = DataSourceConfiguration.wrap(old.to_json())`

and then change any values on `new` that need changing.